core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1018,7 +1025,9 @@ private[spark] class BlockManager(
           try {
             replicate(blockId, bytesToReplicate, level, remoteClassTag)
           } finally {
-            bytesToReplicate.dispose()
+            if (!level.useOffHeap) {
```

Comment:
Ok. I'll see what else is calling that method to validate that a fix there won't break something else, and I'll add a unit test to validate that calling `StorageUtils.dispose` on a direct byte buffer that isn't memory mapped doesn't actually dispose it.